### PR TITLE
create a trip list view

### PR DESCRIPTION
**Pull request by SamYang10086**

### CODE REVIEW

File: Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    // Title
+                    Text("Cities")
+                        .font(.largeTitle)
+                        .bold()
+                        .padding(.top, 40)
+                        .padding(.leading, 20)
+                    
+                    // ScrollView to hold the cards
+                    ScrollView {
+                        VStack(spacing: 20) {
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift:31]: Use LazyVStack in scrollView

File: Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                            ForEach(cities, id: \.cityName) { city in
+                                CityCardView(cityName: city.cityName, countryName: city.countryName, imageName: city.imageName)
+                            }
+                        }
+                        .padding(.horizontal, 20)
+                    }
+                }
+                
+                // Add button in the top-right corner
+                Button(action: {}) {
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift:42]: Button {
//action
}  label: {
}

use this format

File: Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+struct CityCardView: View {
+    var cityName: String
+    var countryName: String
+    var imageName: String
+    
+    var body: some View {
+        ZStack(alignment: .bottomLeading) {
+            Image(imageName)
+                            .resizable() // Makes the image resizable
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift:18]: format

File: Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                            .aspectRatio(contentMode: .fill) // Fills the card
+                            .frame(height: 150)
+                            .cornerRadius(15)
+                            .clipped() // Ensures the image fits within the bounds of the card
+            
+            RoundedRectangle(cornerRadius: 15)
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift:24]: Any other way to do this?
you are using a zstack of 3 layers now

File: Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                .fill(Color.black.opacity(0.3))
+                .frame(height: 150)
+            
+            VStack(alignment: .leading) {
+                Text(cityName)
+                    .font(.title)
+                    .bold()
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift:31]: .fontWeight(.bold)

File: Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                Spacer()
+                
+                // Edit button
+                Button(action: {}) {
+                    Text("Edit")
+                        .font(.caption)
+                        .foregroundColor(.white)
+                        .padding(.horizontal, 10)
+                        .padding(.vertical, 5)
+                        .background(Color.gray)
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift:47]: Color.brandPrimary

**Review by Topia99 — COMMENTED** [review]
Need fix

File: Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift (hunk 1 above)
Comment (SamYang10086) [reply Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift:31]: done

File: Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift (hunk 2 above)
Comment (SamYang10086) [reply Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift:42]: done

File: Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift (hunk 3 above)
Comment (SamYang10086) [reply Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift:18]: done

File: Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift (hunk 4 above)
Comment (SamYang10086) [reply Travel-Planner/Travel-Planner/Views/TripListPageViews/CityCardView.swift:24]: I did another way using overlay

File: Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift (hunk 1 above)
Comment (Topia99) [reply Travel-Planner/Travel-Planner/Screens/TripListView/TripListView.swift:31]: It's still VStack? Did you push your changes?

**Pull request closed without merging**